E0138 needs to be updated to new format

From: src/test/compile-fail/E0138.rs

E0138 needs a span_label, updating it from:

```
error[E0138]: multiple 'start' functions
  --> src/test/compile-fail/E0138.rs:17:1
   |
17 | fn f(argc: isize, argv: *const *const u8) -> isize {} //~ ERROR E0138
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0138]: multiple 'start' functions
  --> src/test/compile-fail/E0138.rs:17:1
   |
17 | fn f(argc: isize, argv: *const *const u8) -> isize {} //~ ERROR E0138
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ multiple `start` functions
```

Bonus: would be better to just underline the name of the function and show the other start function:

```
error[E0138]: multiple 'start' functions
  --> src/test/compile-fail/E0138.rs:17:1
   |
14 | fn foo(argc: isize, argv: *const *const u8) -> isize {}
        --- previous `start` function here 
...
17 | fn f(argc: isize, argv: *const *const u8) -> isize {} //~ ERROR E0138
   |    ^ multiple `start` functions
```

--- a/src/librustc/middle/entry.rs
+++ b/src/librustc/middle/entry.rs
@@ -132,8 +132,13 @@ fn find_item(item: &Item, ctxt: &mut EntryContext, at_root: bool) {
             if ctxt.start_fn.is_none() {
                 ctxt.start_fn = Some((item.id, item.span));
             } else {
-                span_err!(ctxt.session, item.span, E0138,
-                          "multiple 'start' functions");
+                struct_span_err!(
+                    ctxt.session, item.span, E0138,
+                    "multiple 'start' functions")
+                    .span_label(ctxt.start_fn.unwrap().1,
+                                &format!("previous `start` function here"))
+                    .span_label(item.span, &format!("multiple `start` functions"))
+                    .emit();
             }
         },
         EntryPointType::None => ()

--- a/src/test/compile-fail/E0138.rs
+++ b/src/test/compile-fail/E0138.rs
@@ -12,6 +12,9 @@
 
 #[start]
 fn foo(argc: isize, argv: *const *const u8) -> isize {}
+//~^ NOTE previous `start` function here
 
 #[start]
-fn f(argc: isize, argv: *const *const u8) -> isize {} //~ ERROR E0138
+fn f(argc: isize, argv: *const *const u8) -> isize {}
+//~^ ERROR E0138
+//~| NOTE multiple `start` functions

